prettyprinter/src/Data/Text/Prettyprint/Doc/Internal.hs

```diff
@@ -352,7 +361,95 @@ instance Pretty Lazy.Text where pretty = pretty . Lazy.toStrict
 -- []
 instance Pretty Void where pretty = absurd
 
+-- | Overloaded conversion to 'Doc', lifted to unary type constructors.
+--
+-- This is most useful for:
+-- 1. defining 'Pretty' instances for recursive types,
+-- 2. defining 'Pretty' instances for type constructors without 'Functor'
+--    instances, and
+-- 3. efficiently pretty-printing type constructors with 'Functor' instances
+--    whose 'fmap' traverses the whole structure.
+--
+-- Laws:
+--
+--   1. output should be pretty. :-)
```

Comment:
I think there should be an example for an instance definition here, for example the `[]` one. And the law joke doesn’t need to be repeated again ;-)

I think the key to the Pretty1 documentation should be showing how it is useful without going into too much detail. Like »here we define Pretty without using Pretty1, see how Pretty1 makes our lives much easier?«